File: tests/machine/x/fs/cross_join_filter.fs
open System

type Anon1 = {
    n: int
    l: string
}
let nums: int list = [1; 2; 3]
let letters: string list = ["A"; "B"]
let pairs: obj list = [ for n in nums do 
  for l in letters do if n % 2 = 0 then yield { n = n; l = l } ]
printfn "%s" "--- Even pairs ---"
for p in pairs do
    printfn "%s" (String.concat " " [string p.n; string p.l])

File: tests/transpiler/x/pas/cross_join_filter.pas
{$mode objfpc}
program Main;
type Anon4 = record
  n: integer;
  l: string;
end;
var
  nums: array of integer;
  letters: array of string;
  pairs: array of Anon4;
  n: integer;
  l: string;
  p: integer;
begin
  nums := [1, 2, 3];
  letters := ['A', 'B'];
  pairs := [];
  for n in nums do begin
  for l in letters do begin
  if (n mod 2) = 0 then begin
  pairs := concat(pairs, [(n: n; l: l)]);
end;
end;
end;
  writeln('--- Even pairs ---');
  for p in pairs do begin
  writeln(p.n, p.l);
end;
end.
